flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -628,9 +692,10 @@ public void open(FileInputSplit fileSplit) throws IOException {
 	 * @see org.apache.flink.api.common.io.InputStreamFSInputWrapper
 	 */
 	protected FSDataInputStream decorateInputStream(FSDataInputStream inputStream, FileInputSplit fileSplit) throws Throwable {
-		// Wrap stream in a extracting (decompressing) stream if file ends with .deflate.
-		if (fileSplit.getPath().getName().endsWith(DEFLATE_SUFFIX)) {
-			return new InflaterInputStreamFSInputWrapper(stream);
+		// Wrap stream in a extracting (decompressing) stream if file ends with a known compression file extension.
+		InflaterInputStreamFactory<?> inflaterInputStreamFactory = getInflaterInputStreamFactory(fileSplit.getPath());
+		if (inflaterInputStreamFactory != null) {
+			return new InputStreamFSInputWrapper(inflaterInputStreamFactory.create(stream));
```

Comment:
It might also be the case that the stream was not compressed at all. It would of course be nice to react appropriately to a missing codec, but how would we know if the current input split belongs to an uncompressed file or a compressed file with an unknown codec?